### PR TITLE
Define library

### PR DESCRIPTION
**Pull request by ViviYe**
implements issue #114 
* Libraries can be defined in futil files
* -library flag now defines the directory to look for library files
* The parser is still being worked on to make import statement optional

### CODE REVIEW

**Comment by kwf37** [comment]
If the -library option is not provided, the compiler will search in the working directory for library files

File: calyx/src/lang/context.rs [hunk 1]
```diff
@@ -210,6 +224,8 @@ impl Into<ast::NamespaceDef> for Context {
         ast::NamespaceDef {
             name: name.into(),
             components,
+            ///TODO: fix it later
```
Comment (rachitnigam) [inline calyx/src/lang/context.rs:227]: TODO comments should make sense to other collaborators! Please follow the [Google Styleguide](https://google.github.io/styleguide/cppguide.html#TODO_Comments) for this.

File: calyx/src/cmdline.rs [hunk 2]
```diff
@@ -23,8 +23,8 @@ pub struct Opts {
     pub file: Option<PathBuf>,
 
     /// Path to the primitives library.
-    #[structopt(long, short, required_unless = "list-passes")]
-    pub libraries: Vec<PathBuf>,
+    #[structopt(long, short)]
+    pub lib_path: Option<PathBuf>,
```
Comment (rachitnigam) [inline calyx/src/cmdline.rs:27]: If possible, make it so that when this is not specified, it defaults to the current directory.

Comment (sgpthomas) [reply calyx/src/cmdline.rs:27]: :+1: You should make it just be a `PathBuf` and then use the `default_value` flag to set the default value to the current directory

**Comment by rachitnigam** [comment]
@ViviYe @kwf37 can you ping us when the changes are done? Re-request the review or something?

File: calyx/src/cmdline.rs (hunk 2 above)
Comment (ViviYe) [reply calyx/src/cmdline.rs:27]: I have a question.. can we put a filePath as a default value? I am trying to use std::env::current_dir() to get the current directory and use that as default and the compiler yells at me for excepecting a string reference...

Comment (rachitnigam) [reply calyx/src/cmdline.rs:27]: I don’t think you can put a pathbuf there. Putting the string “.” Should work.

**Comment by ViviYe** [comment]
changed the option<filepath> to just filepath. The Latency Insensitive test seem to fail half of the times?

**Comment by sgpthomas** [comment]
Weird...i'll check this out

**Comment by sgpthomas** [comment]
ahh, looks like nondeterminstic printing. always a fun time. This should be fixed in master. Looks like git could automatically merge in master, so I let it do that.

**Comment by sgpthomas** [comment]
Looks like some tests from master failed because they don't have `import` statements

**Comment by ViviYe** [comment]
All tests pass now : D

**Comment by sgpthomas** [comment]
Yay!! You can press the merge button now!

**Pull request merged**